lib/fizzy/cxx20/bit.hpp

```diff
@@ -0,0 +1,26 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <type_traits>
+
+namespace fizzy
+{
+/// The non-constexpr implementation of C++20's std::bit_cast.
+/// See https://en.cppreference.com/w/cpp/numeric/bit_cast.
+template <class To, class From>
+[[nodiscard]] inline
+    typename std::enable_if_t<sizeof(To) == sizeof(From) && std::is_trivially_copyable_v<From> &&
+                                  std::is_trivially_copyable_v<To>,
+        To>
+    bit_cast(const From& src) noexcept
+{
+    static_assert(std::is_trivially_constructible_v<To>);  // Additional, non-standard requirement.
```

Comment:
https://en.cppreference.com/w/cpp/numeric/bit_cast#Example has actually this line, do you mean that the final standard does not require it? I guess it has no issues as long as we keep compiling with <C++20 any such usage will be detected in our codebase, if we want to avoid such usage.

Reply:
Yes. The C++20 version only requires the conditions from `std::enable_if`. Personally, I would just use `static_assert` instead of `std::enable_if`, but if C++20 defines it like that let it be. I would also omit the `is_trivially_constructible_v` check, but being more pedantic is ok I guess.